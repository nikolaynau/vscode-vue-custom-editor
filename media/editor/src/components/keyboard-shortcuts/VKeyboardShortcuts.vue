<script setup lang="ts">
import { computed, toRefs } from 'vue';
import VKeybindingLabel from '../keybinding-label/VKeybindingLabel.vue';

export interface KeyItem {
  title: string;
  key: string;
}

export interface Props {
  items?: Array<KeyItem>;
  labelAlign?: 'left' | 'right';
}

const props = withDefaults(defineProps<Props>(), {
  items: () => [],
  labelAlign: 'left'
});

const { labelAlign } = toRefs(props);

const cssClasses = computed(() => ({
  'v-keyboard-shortcuts--label-left': labelAlign.value === 'left',
  'v-keyboard-shortcuts--label-right': labelAlign.value === 'right'
}));
</script>

<template>
  <div class="v-keyboard-shortcuts" :class="cssClasses">
    <dl v-for="(item, index) in items" :key="index">
      <dt>{{ item.title }}</dt>
      <dd>
        <VKeybindingLabel :value="item.key" />
      </dd>
    </dl>
  </div>
</template>

<style>
@import 'keyboard-shortcuts';
</style>
