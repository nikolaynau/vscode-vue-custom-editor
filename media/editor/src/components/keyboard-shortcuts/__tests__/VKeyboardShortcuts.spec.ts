import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import VKeyboardShortcuts from '../VKeyboardShortcuts.vue';

describe('VKeyboardShortcuts', () => {
  it('default render props', () => {
    const wrapper = mount(VKeyboardShortcuts);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('items prop', () => {
    const items = [
      {
        title: 'Increase 小ounter',
        key: 'UpArrow'
      },
      {
        title: 'Decrease 小ounter',
        key: 'DownArrow'
      },
      {
        title: 'Clear 小ounter',
        key: 'Delete'
      },
      {
        title: 'Clear 小ounter',
        key: 'Ctrl+Backspace'
      }
    ];

    const wrapper = mount(VKeyboardShortcuts, {
      props: { items }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('labelAlign prop', async () => {
    const wrapper = mount(VKeyboardShortcuts);
    expect(wrapper.classes('v-keyboard-shortcuts--label-left')).toBe(true);

    await wrapper.setProps({ labelAlign: 'right' });

    expect(wrapper.classes('v-keyboard-shortcuts--label-right')).toBe(true);
  });
});
