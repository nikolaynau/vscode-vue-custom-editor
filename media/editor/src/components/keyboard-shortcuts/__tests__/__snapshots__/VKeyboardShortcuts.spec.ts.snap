// Vitest Snapshot v1

exports[`VKeyboardShortcuts > default render props 1`] = `"<div class=\\"v-keyboard-shortcuts v-keyboard-shortcuts--label-left\\"></div>"`;

exports[`VKeyboardShortcuts > items prop 1`] = `
"<div class=\\"v-keyboard-shortcuts v-keyboard-shortcuts--label-left\\">
  <dl>
    <dt>Increase 小ounter</dt>
    <dd>
      <div class=\\"v-keybinding-label\\" title=\\"UpArrow\\">
        <!--v-if--><span class=\\"v-keybinding-label__key\\">UpArrow</span>
      </div>
    </dd>
  </dl>
  <dl>
    <dt>Decrease 小ounter</dt>
    <dd>
      <div class=\\"v-keybinding-label\\" title=\\"DownArrow\\">
        <!--v-if--><span class=\\"v-keybinding-label__key\\">DownArrow</span>
      </div>
    </dd>
  </dl>
  <dl>
    <dt>Clear 小ounter</dt>
    <dd>
      <div class=\\"v-keybinding-label\\" title=\\"Delete\\">
        <!--v-if--><span class=\\"v-keybinding-label__key\\">Delete</span>
      </div>
    </dd>
  </dl>
  <dl>
    <dt>Clear 小ounter</dt>
    <dd>
      <div class=\\"v-keybinding-label\\" title=\\"Ctrl+Backspace\\">
        <!--v-if--><span class=\\"v-keybinding-label__key\\">Ctrl</span><span class=\\"v-keybinding-label__separator\\">+</span><span class=\\"v-keybinding-label__key\\">Backspace</span>
      </div>
    </dd>
  </dl>
</div>"
`;
